modules/exploits/windows/http/manageengine_adaudit_plus_cve_2022_28219.rb

```diff
@@ -0,0 +1,393 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::Remote::TcpServer
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::JavaDeserialization
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ManageEngine ADAudit Plus CVE-2022-28219',
+        'Description' => %q{
+          This module exploits CVE-2022-28219, which is a pair of
+          vulnerabilities in ManageEngine ADAudit Plus versions before build
+          7060: a path traversal in the /cewolf endpoint, and a blind XXE in,
+          to upload and execute an executable file.
+        },
+        'Author' => [
+          'Naveen Sunkavally', # Initial PoC + disclosure
+          'Ron Bowes', # Analysis and module
+        ],
+        'References' => [
+          ['CVE', '2022-28219'],
+          ['URL', 'https://www.horizon3.ai/red-team-blog-cve-2022-28219/'],
+          ['URL', 'https://attackerkb.com/topics/Zx3qJlmRGY/cve-2022-28219/rapid7-analysis'],
+          ['URL', 'https://www.manageengine.com/products/active-directory-audit/cve-2022-28219.html'],
+        ],
+        'DisclosureDate' => '2022-06-29',
+        'License' => MSF_LICENSE,
+        'Platform' => 'java',
+        'Arch' => ARCH_JAVA,
```

Comment:
```suggestion
        'Arch' => [ ARCH_JAVA, ARCH_CMD ],
```

??

Reply:
Do we actually need ARCH_JAVA? I copied that from another exploit

Reply:
We actually wondered about this, if it exploit cannot launch a payload that is a java jar then just having `ARCH_CMD` would be more correct.

Reply:
Cool, I removed that